### PR TITLE
Tooltip alignment options, a11y improvements

### PR DESCRIPTION
**Pull request by pixelbandito**
Addresses issues #305 and #298 

* Adds an optional secondary dimension to Tooltip config, e.g. `top-left`.
* Uses `CSSTransition` to unmount tooltips when they're closed and clean up fade in / fade out.
* a11y improvements
  * "hover" tooltips are also controlled with focus/blur
  * users can dismiss tooltips (even hover/focus ones) with the Escape key
  * you can mouse from a tooltip trigger to the tooltip content in hover mode without it disappearing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,8 +48,8 @@
     "pre-commit:lint": "yarn lint-staged",
     "pre-commit:types": "yarn tsc",
     "pre-commit:test": "yarn test",
-    "test": "cross-env CI=1 react-scripts test --env=jsdom",
-    "test:watch": "react-scripts test --env=jsdom",
+    "test": "cross-env CI=1 react-scripts test --env=jest-environment-jsdom-sixteen",
```
Comment (pixelbandito) [inline package.json:51]: This was necessary for some tests to work with hover / focus.

File: package.json [hunk 2]
```diff
@@ -88,10 +88,10 @@
     "@storybook/addons": "^5.3.19",
     "@storybook/preset-create-react-app": "^3.0.0",
     "@storybook/react": "^5.3.19",
-    "@testing-library/jest-dom": "^5.10.1",
-    "@testing-library/react": "^10.2.1",
-    "@testing-library/react-hooks": "^3.3.0",
-    "@testing-library/user-event": "^12.0.7",
+    "@testing-library/jest-dom": "5.11.9",
```
Comment (pixelbandito) [inline package.json:91]: This stuff was just cleanup, trying to fix an issue that ended up being unrelated.

File: src/components/Tooltip/story.tsx [hunk 3]
```diff
@@ -62,74 +54,62 @@ const Element: React.FC<React.HTMLAttributes<HTMLDivElement>> = (props) => (
   <div className="mr-5" {...props} />
 );
 
-const Overview = () => {
```
Comment (pixelbandito) [inline src/components/Tooltip/story.tsx:65]: When this component was defined outside the story's render function, knobs didn't work.

**Review by mikemfleming — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (pixelbandito) [reply package.json]: Nerp, after I fixed some other stuff, I didn't need jsdom 16 after all. And including it broke backwards compatibility with Node 10 (which we say we support).
So I removed it, but I'm kinda sad that Node 10 support might stop us from upgrading stuff, in general.

**Review by bkonuwa — APPROVED** [review]

**Pull request merged**